### PR TITLE
Use ColumnOptions to track report columns

### PR DESCRIPTION
**Pull request by NoahCarnahan**
(this PR is *not* against master)

@kaapstorm 
This addresses the "Generate column IDs to allow columns to appear multiple times" item, and also cleans up a couple things. I'll look at the merge conflict tomorrow AM.

### CODE REVIEW

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -634,52 +661,52 @@ class ConfigureReport(ReportBuilderView):
...
-    #
-    # def get_columns(self):
-    #     columns = self.report_column_options
-    #     return columns.values()
+
+    def get_columns(self):
+        columns = self.report_column_options
+        # TODO: All the fields are called different things. Let's change it on the front end eventually,
+        #       but re-map for now to see if its working
+        def remap_fields(column):
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E301 expected 1 blank line, found 0

File: corehq/apps/userreports/views.py [hunk 2]
```diff
@@ -799,17 +828,16 @@ def get_builder_report_type(report_data_):
...
-            aggregation_columns = []
+
         try:
             report_configuration = ReportConfiguration(
                 domain=app.domain,
                 config_id=data_source_config_id,
                 title=report_name,
-                aggregation_columns=aggregation_columns,
-                columns=[to_report_column(c, i) for i, c in enumerate(report_data['columns'])],
+                aggregation_columns=_get_aggregation_columns(report_data['aggregate'], report_data['columns'], self.report_column_options),
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E501 line too long (139 > 115 characters)

File: corehq/apps/userreports/views.py [hunk 3]
```diff
@@ -799,17 +828,16 @@ def get_builder_report_type(report_data_):
...
+                columns=list(chain.from_iterable(
+                    to_report_columns(c, i, self.report_column_options) for i, c in enumerate(report_data['columns'])
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/userreports/views.py [hunk 4]
```diff
@@ -851,17 +879,19 @@ class ReportPreview(BaseDomainView):
...
+        )
+
         table = ConfigurableReport.report_config_table(
             domain=domain,
             config_id=data_source,
             title='{}_{}_{}'.format(TEMP_REPORT_PREFIX, domain, data_source),
             description='',
-            aggregation_columns=aggregation_columns,
-            columns=[to_report_column(c, i) for i, c in enumerate(report_data['columns'])],
+            aggregation_columns=_get_aggregation_columns(report_data['aggregate'], report_data['columns'], column_options),
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E501 line too long (123 > 115 characters)

**Comment by kaapstorm** [comment]
@NoahCarnahan this is great! The conflict is mostly from the first commit, because I worked on the same problem (but used four long-winded getters instead of one nice short method).

File: corehq/apps/userreports/views.py [hunk 5]
```diff
@@ -834,30 +829,24 @@ def get_builder_report_type(report_data_):
...
         try:
             report_configuration = ReportConfiguration(
-                domain=app.domain,
+                domain=self.domain,
                 config_id=data_source_config_id,
-                title=report_name,
-                aggregation_columns=aggregation_columns,
-                columns=[to_report_column(c, i) for i, c in enumerate(report_data['columns'])],
+                title=self._get_report_name(request),
+                aggregation_columns=_get_aggregation_columns(report_data['aggregate'], report_data['columns'], self.report_column_options),
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E501 line too long (139 > 115 characters)

File: corehq/apps/userreports/views.py [hunk 6]
```diff
@@ -834,30 +829,24 @@ def get_builder_report_type(report_data_):
...
+                columns=list(chain.from_iterable(
+                    to_report_columns(c, i, self.report_column_options) for i, c in enumerate(report_data['columns'])
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/userreports/views.py [hunk 7]
```diff
@@ -834,30 +829,26 @@ def get_builder_report_type(report_data_):
...
                 config_id=data_source_config_id,
-                title=report_name,
-                aggregation_columns=aggregation_columns,
-                columns=[to_report_column(c, i) for i, c in enumerate(report_data['columns'])],
+                title=self._get_report_name(request),
+                aggregation_columns=_get_aggregation_columns(
+                    report_data['aggregate'], report_data['columns'], self.report_column_options
+                ),
+                columns=list(chain.from_iterable(
+                    to_report_columns(c, i, self.report_column_options) for i, c in enumerate(report_data['columns'])
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E501 line too long (117 > 115 characters)

**Comment by NoahCarnahan** [comment]
👍 easy to fix the conflict

**Pull request merged**